client/driver/executor/executor_linux.go

```diff
@@ -41,90 +33,89 @@ var (
 	}
 )
 
-func NewExecutor(ctx *ExecutorContext) Executor {
-	return NewLinuxExecutor(ctx)
-}
-
-func NewLinuxExecutor(ctx *ExecutorContext) Executor {
-	return &LinuxExecutor{ExecutorContext: ctx}
-}
-
-// Linux executor is designed to run on linux kernel 2.8+.
-type LinuxExecutor struct {
-	*ExecutorContext
-	cmd  exec.Cmd
-	user *user.User
-	l    sync.Mutex
-
-	// Isolation configurations.
-	groups   *cgroupConfig.Cgroup
-	taskName string
-	taskDir  string
-	allocDir string
-
-	// Spawn process.
-	spawn *spawn.Spawner
-}
+// configureIsolation configures chroot and creates cgroups
+func (e *UniversalExecutor) configureIsolation() error {
+	if e.ctx.FSIsolation {
+		if err := e.configureChroot(); err != nil {
+			return err
+		}
+	}
 
-func (e *LinuxExecutor) Command() *exec.Cmd {
-	return &e.cmd
+	if e.ctx.ResourceLimits {
+		if err := e.configureCgroups(e.ctx.TaskResources); err != nil {
+			return fmt.Errorf("error creating cgroups: %v", err)
+		}
+		if err := e.applyLimits(os.Getpid()); err != nil {
+			if er := e.destroyCgroup(); er != nil {
```

Comment:
@dadgar If there is an error in configuring either the cgroups, chroot or applying the cgroup to the plugin process, we shouldn't continue any further and exit early.